action.yml

```diff
@@ -4,37 +4,52 @@ branding:
   icon: "image"
   color: "orange"
 inputs:
-  github-token:
-    description: "GitHub token"
-    required: true
   pizza-args:
     description: "OpenSauced Pizza CLI command to run"
     default: "generate codeowners ./"
     required: true
+  commit-and-push:
+    description: "Commit and push changes"
+    default: "false"
+    required: false
+
 runs:
   using: "composite"
   steps:
     - name: Checkout
       uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
 
-    - name: Setup Node
-      uses: actions/setup-node@v3
+    - name: Setup Go
+      uses: actions/setup-go@v4
       with:
-        node-version: "20"
+        go-version: "1.22.5"
+
+    - name: Install pizza CLI
+      run: |
+        go install github.com/open-sauced/pizza-cli@beta
+      shell: bash
 
-    - name: Install dependencies
-      run: npm ci
+    - name: Check pizza CLI version
+      run: pizza-cli version
       shell: bash
-      working-directory: ${{ github.action_path }}
 
-    - name: Install pizza-cli
-      run: go install github.com/open-sauced/pizza-cli@latest
+    - name: Run pizza CLI command
+      run: |
+        pwd
+        ls -la
+        pizza-cli ${{ inputs.pizza-args }}
       shell: bash
-      working-directory: ${{ github.action_path }}
+      working-directory: ${{ github.workspace }}
 
-    - name: Run action
-      run: node ${{ github.action_path }}/index.js
+    - name: Commit and push changes
+      if: ${{ inputs.commit-and-push == 'true' }}
+      run: |
+        git config user.name github-actions
```

Comment:
This is minimal boilerplate for committing and pushing so I kept it inline the action.yml. It's also an optional step.